src/tests/tests/yellowlist.js

```diff
@@ -0,0 +1,116 @@
+/* globals badger:false */
+
+(function() {
+
+  function get_ylist() {
+    return badger.storage.getBadgerStorageObject(
+      'cookieblock_list').getItemClones();
+  }
+
+  let constants = require('constants');
+
+  // fake server to simulate XMLHttpRequests
+  let server;
+
+  QUnit.module("Yellowlist", {
+    before: (/*assert*/) => {
+      server = sinon.fakeServer.create({
+        respondImmediately: true
+      });
+    },
+
+    after: (/*assert*/) => {
+      server.restore();
+    }
+  });
+
+  QUnit.test("Updating to a valid list", (assert) => {
+    let done = assert.async();
+    assert.expect(3);
+
+    let ylist = get_ylist();
+    assert.ok(!!Object.keys(ylist).length, "Yellowlist is not empty");
+
+    // remove a domain
+    let removed_domain = Object.keys(ylist)[0];
+    delete ylist[removed_domain];
+
+    // add a domain
+    const NEW_YLIST_DOMAIN = "widgets.example.com";
+    ylist[NEW_YLIST_DOMAIN] = true;
+
+    // respond with current list plus new domain
+    server.respondWith("GET", constants.YELLOWLIST_URL,
+      [200, {}, Object.keys(ylist).join("\n")]);
+
+    badger.updateCookieBlockList(function (success) {
+      assert.ok(success, "Callback status indicates success");
+      assert.deepEqual(get_ylist(), ylist, "List got updated");
+      done();
+    });
```

Comment:
I don't think we should restore the domain removal logic. Do we really need to test right now that we don't do anything outside of removing the domain from the cookieblock list here? I don't think so ...